realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -103,8 +103,13 @@ public void removeFromRealm() {
      * @return {@code true} if the object is still accessible, {@code false} otherwise or if it is a
      * standalone object.
      */
-    public boolean isValid() {
-        return row != null && row.isAttached();
+    public boolean
+    isValid() {
+        try {
+            return row != null && row.isAttached();
+        } catch (IllegalStateException e) {
+            return false;
```

Comment:
Well, yes and no. The description could perhaps be better. The purpose of it is to act as a replacement when the row accessor is no longer available. In that context returning false for `isAttached` makes sense IMO.